src/components/AlertModal/AlertModal.tsx

```diff
@@ -0,0 +1,165 @@
+    flexDirection: "row",
+    marginHorizontal: 16,
+    marginBottom: 24
+  },
+  modalSecondaryBtm: {
+    marginRight: 8,
+    flexShrink: 1
+  },
+  modalPrimaryButton: {
+    flex: 1
+  }
+});
+type AlertType = "ERROR" | "WARN" | "CONFIRM" | "INFO";
+
+// Add on the collection of action button texts
+// try to declare object as { [key:string]: CallToActionButtonTexts }
+// but somehow CallToActionKey unable to determine the keys on compile time
+const callToActionCollection = Object.freeze({
+  YES_NO: { primaryActionText: "YES", secondaryActionText: "NO" },
+  OK_CANCEL: { primaryActionText: "OK", secondaryActionText: "CANCEL" },
+  CONFIRM_CANCEL: {
+    primaryActionText: "CONFIRM",
+    secondaryActionText: "CANCEL"
+  }
+});
+
+interface CallToActionButtonTexts {
+  primaryActionText: string;
+  secondaryActionText?: string;
+}
+
+export type CallToActionKey = keyof typeof callToActionCollection;
+
+export interface AlertModalProp {
+  alertType: AlertType;
+  title: string;
+  description: string;
+  buttonTextType: CallToActionKey;
+  visible: boolean;
+  onOk: () => void;
+  onCancel?: () => void;
+  onExit?: any;
+}
+
+export const AlertModal: FunctionComponent<AlertModalProp> = (
+  props: AlertModalProp
+) => {
+  const callToAction = useMemo(
+    () =>
+      callToActionCollection[props.buttonTextType] as CallToActionButtonTexts,
+    [props.buttonTextType]
+  );
+
+  const primaryButton = useMemo(() => {
+    switch (props.alertType) {
+      case "WARN":
+        return (
+          <DangerButton
+            text={callToAction.primaryActionText}
+            fullWidth={true}
+            onPress={() => {
+              props.onExit();
+              props.onCancel && props.onCancel();
+            }}
+          />
+        );
+      default:
+        return (
+          <DarkButton
+            text={callToAction.primaryActionText}
+            fullWidth={true}
+            onPress={() => {
+              props.onExit();
+              props.onCancel && props.onCancel();
+            }}
+          />
+        );
+    }
+  }, [callToAction.primaryActionText, props]);
+
+  return (
+    <Modal animationType="fade" transparent={true} visible={props.visible}>
+      <View style={styles.centeredView}>
+        <View style={styles.modalView}>
+          {props.alertType === "ERROR" && (
+            <AlertLogo style={styles.alertIcon} />
+          )}
+          <Text style={styles.modalTitle}>{props.title}</Text>
+          <Text style={styles.modalText}>{props.description}</Text>
+          <View style={styles.modalGroupButton}>
+            {props.alertType !== "ERROR" && props.alertType !== "INFO" && (
+              <View style={styles.modalSecondaryBtm}>
+                <SecondaryButton
+                  text={callToAction.secondaryActionText || ""}
+                  fullWidth={true}
+                  onPress={() => {
+                    props.onExit();
+                    props.onCancel && props.onCancel();
```

Comment:
Some new typescript shortcuts
```suggestion
                    props.onCancel?.();
```

Reply:
Niceeee 👍 . Updated